src/components/logs/Context.tsx

```diff
@@ -0,0 +1,174 @@
+const MAX_INTERVAL = 1000; //  we don't want to wait too longer between pings
+const LogsContext = createContext<LogsContextValue | null>(null);
+
+const LogsContextProvider = ({
+    children,
+    token,
+    disableIntervalFetching,
+    fetchAll,
+}: Props) => {
+    const supabaseClient = useClient();
+
+    // Privates
+    const allLogs = useRef<ViewLogs_Line[]>([]);
+    const offset = useRef(0);
+    const timeoutRef = useRef<number | undefined>();
+    const interval = useRef(DEFAULT_POLLING_INTERVAL / 2);
+    const emptyResponses = useRef(
+        disableIntervalFetching ? MAX_EMPTY_CALLS : 0
+    );
+
+    // Publics
+    const [logs, setLogs] = useState<ViewLogs_Line[]>([]);
+    const [stopped, setStopped] = useState(false);
+    const [networkFailure, setNetworkFailure] = useState(false);
+
+    const fetchLogs = async (offsetVal: number) => {
+        const queryParams = {
+            bearer_token: token,
+        };
+        if (fetchAll) {
+            return supabaseClient
+                .rpc<ViewLogs_Line>(RPCS.VIEW_LOGS, queryParams)
+                .throwOnError();
+        } else {
+            return supabaseClient
+                .rpc<ViewLogs_Line>(RPCS.VIEW_LOGS, queryParams)
+                .throwOnError()
+                .range(offsetVal, offsetVal + 10);
+        }
+    };
+
+    const stopLogs = (networkFailed?: boolean) => {
+        timeoutCleanUp(timeoutRef.current);
+        setStopped(true);
+        emptyResponses.current = emptyResponses.current + 1;
+        if (networkFailed) {
+            setNetworkFailure(true);
+        }
+    };
+
+    const start = () => {
+        if (!token) return;
+
+        const makeApiCall = () => {
+            return fetchLogs(offset.current).then(
+                (fetchLogsResponse) => {
+                    timeoutCleanUp(timeoutRef.current);
+
+                    if (fetchLogsResponse.error) {
+                        stopLogs(true);
+                    } else {
+                        setNetworkFailure(false);
+
+                        const { data: viewLogsResponse } = fetchLogsResponse;
+                        if (hasLength(viewLogsResponse)) {
+                            offset.current =
+                                offset.current + viewLogsResponse.length;
+                            allLogs.current =
+                                allLogs.current.concat(viewLogsResponse);
+                            setLogs(allLogs.current);
+                        } else {
+                            emptyResponses.current = emptyResponses.current + 1;
+                        }
+
+                        if (
+                            !disableIntervalFetching &&
+                            MAX_EMPTY_CALLS >= emptyResponses.current
+                        ) {
+                            interval.current = incrementInterval(
+                                interval.current,
+                                MAX_INTERVAL
+                            );
```

Comment:
This is one change form the ole `useInterval` approach. We can not back off just a little bit.